src/coreclr/gc/vxsort/isa_detection.cpp

```diff
@@ -66,36 +78,44 @@ SupportedISA DetermineSupportedISA()
         (xcr0 & 0xe6) == 0xe6 &&
         (FeatureMask & (XSTATE_MASK_AVX | XSTATE_MASK_AVX512)) == (XSTATE_MASK_AVX | XSTATE_MASK_AVX512))
     {
-        return (SupportedISA)((int)SupportedISA::AVX2 | (int)SupportedISA::AVX512F);
+        IsaFlags |= (int)SupportedISA::AVX512F;
     }
 
     // check if AVX2 is supported by both processor and OS
     if ((reg[EBX] & AVX2) &&
         (xcr0 & 0x06) == 0x06 &&
         (FeatureMask & XSTATE_MASK_AVX) == XSTATE_MASK_AVX)
     {
-        return SupportedISA::AVX2;
+        IsaFlags |= (int)SupportedISA::AVX2;
     }
 
-    return SupportedISA::None;
+    return (SupportedISA)IsaFlags;
 }
 
 #elif defined(TARGET_UNIX)
 
 SupportedISA DetermineSupportedISA()
 {
     __builtin_cpu_init();
-    if (__builtin_cpu_supports("avx2"))
+
+    int IsaFlags = 0;
+
+    if (__builtin_cpu_supports("apx"))
     {
-        if (__builtin_cpu_supports("avx512f"))
-            return (SupportedISA)((int)SupportedISA::AVX2 | (int)SupportedISA::AVX512F);
-        else
-            return SupportedISA::AVX2;
+        IsaFlags |= (int)SupportedISA::APX;
     }
-    else
+
+    if (__builtin_cpu_supports("avx512"))
     {
-        return SupportedISA::None;
+        IsaFlags |= (int)SupportedISA::APX;
     }
+
+    if (__builtin_cpu_supports("avx2"))
+    {
+        IsaFlags |= (int)SupportedISA::APX;
```

Comment:
It is not necessary to detect APX here until the APX-specific implementation of the GC sort gets added.